src/main/kotlin/no/nav/familie/ef/sak/task/GrensesnittavstemmingTask.kt

```diff
@@ -19,23 +22,41 @@ class GrensesnittavstemmingTask(private val avstemmingService: AvstemmingService
     val logger: Logger = LoggerFactory.getLogger(this.javaClass)
 
     override fun doTask(task: Task) {
-        val fraTidspunkt = LocalDate.parse(task.payload).atStartOfDay()
-        val tilTidspunkt = task.triggerTid.toLocalDate().atStartOfDay()
+        val payload = objectMapper.readValue<GrensesnittavstemmingPayload>(task.payload)
 
-        logger.info("Gjør avstemming mot oppdrag fra $fraTidspunkt til ${task.triggerTid.toLocalDate()}")
+        val fraTidspunkt = payload.fraDato.atStartOfDay()
+        val tilTidspunkt = task.triggerTid.toLocalDate().atStartOfDay()
 
-        avstemmingService.grensesnittavstemOppdrag(fraTidspunkt, tilTidspunkt)
+        logger.info("Gjør ${task.id} ${payload.stønadstype} avstemming mot oppdrag fra $fraTidspunkt til $tilTidspunkt")
 
+        when (payload.stønadstype) {
+            Stønadstype.OVERGANGSSTØNAD -> avstemmingService.grensesnittavstemOvergangsstønad(fraTidspunkt, tilTidspunkt)
+            else -> throw Error("Grensesnittavstemming for ${payload.stønadstype} er ikke implementert")
+        }
     }
 
     override fun onCompletion(task: Task) {
         val nesteVirkedag: LocalDate = VirkedagerProvider.nesteVirkedag(task.triggerTid.toLocalDate())
 
+        opprettNyTask(task.triggerTid.toLocalDate(), nesteVirkedag)
+    }
+
+    fun utløsGrensesnittavstemming(fraDato: LocalDate, stønadstype: Stønadstype, triggerTid: LocalDate?): Task {
+        val nesteVirkedag: LocalDate = triggerTid ?: VirkedagerProvider.nesteVirkedag(fraDato)
+
+        return opprettNyTask(fraDato, nesteVirkedag, stønadstype)
+    }
+
+    private fun opprettNyTask(fraDato: LocalDate,
+                              nesteVirkedag: LocalDate,
+                              stønadstype: Stønadstype = Stønadstype.OVERGANGSSTØNAD): Task {
```

Comment:
`stønadstype: Stønadstype = Stønadstype.OVERGANGSSTØNAD`
Syns jeg er litt skummel, hvis man legger til flere stønadstyper senere og ikke husker på å endre denne så kommer man opprette en ny task for overgangsstønad, blir sikkert fanget opp noen gang og gir kanskje ikke så store konsekvenser?

Reply:
Gott poæng, fikset det her https://github.com/navikt/familie-ef-sak/pull/356/commits/37fb4d09ce671b67163e8ed28d0d769fcff29621